### PR TITLE
Continued Implementation of Ratetimeintransit and Shoptimeintransit requests. See #145

### PR DESCRIPTION
**Pull request by ryross**
This PR includes the work of @mazzarito in #145. 

The RateTimeInTransit Response sends BusinessDaysInTransit instead of BusinessTransitDays in the EstimatedArrival object so I got rid of the inheritance that @mazzarito added opted to use a trait instead. If we kept the inheritance, we could have had getBusinessDaysInTransit() and getBusinessTransitDays(). Obviously not ideal. 

@mazzarito, it might be cleaner if I submit a PR on your fork, and then you update the original PR. @stefandoorn Let me know if you would prefer that.

### CODE REVIEW

**Comment by stefandoorn** [comment]
Submitting the PR to the fork has my preference indeed, afterward the branch can be rebased to look a bit more clean.

**Comment by ryross** [comment]
10-4 @stefandoorn. Created the PR here: https://github.com/mazzarito/php-ups-api/pull/1

**Comment by stefandoorn** [comment]
:-) If it takes long, please re-open and we'll see how to handle it.

**Comment by ryross** [comment]
K. In the mean time, would you take a look at the changes I've made?

File: src/Entity/EstimatedArrival.php [hunk 1]
```diff
@@ -42,27 +39,89 @@ public function __construct(\stdClass $response = null)
...
+    /**
+     * @param mixed $BusinessTransitDays
+     */
+    public function setBusinessTransitDays($BusinessTransitDays)
+    {
+        $this->BusinessTransitDays = $BusinessTransitDays;
+    }
+
+    /**
+     * @return mixed
```
Comment (stefandoorn) [inline src/Entity/EstimatedArrival.php]: Isn't this always a string?

File: src/Entity/EstimatedArrival.php [hunk 2]
```diff
@@ -42,27 +39,89 @@ public function __construct(\stdClass $response = null)
...
+     */
+    public function setTime($Time)
+    {
+        $this->Time = $Time;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getPickupDate()
```
Comment (stefandoorn) [inline src/Entity/EstimatedArrival.php]: Same here, isn't it a string? Or null?

File: src/Entity/EstimatedArrival.php [hunk 3]
```diff
@@ -42,27 +39,89 @@ public function __construct(\stdClass $response = null)
...
+    /**
+     * @param mixed $PickupDate
+     */
+    public function setPickupDate($PickupDate)
+    {
+        $this->PickupDate = $PickupDate;
+    }
+
+    /**
+     * @return mixed
```
Comment (stefandoorn) [inline src/Entity/EstimatedArrival.php]: Same here.

File: src/Entity/EstimatedArrival.php [hunk 4]
```diff
@@ -42,27 +39,89 @@ public function __construct(\stdClass $response = null)
...
+     */
+    public function setPickupTime($PickupTime)
+    {
+        $this->PickupTime = $PickupTime;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getDate()
```
Comment (stefandoorn) [inline src/Entity/EstimatedArrival.php]: Same here.

File: src/Entity/RateTimeInTransit/EstimatedArrival.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    protected $businessDaysInTransit;
+
+    /**
+     * @param \stdClass|null $response
+     */
+    public function __construct(\stdClass $response = null)
+    {
+        if (null !== $response) {
+            self::build($response);
```
Comment (stefandoorn) [inline src/Entity/RateTimeInTransit/EstimatedArrival.php]: I'm not 100% sure if this is correct as the trait defines it as non-static. Didn't you mean `$this->build($response)`? Same for some other locations.

File: src/Entity/EstimatedArrivalTrait.php [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    /**
+     * @param Pickup $Pickup
+     */
+    public function setPickup($Pickup)
+    {
+        $this->Pickup = $Pickup;
+    }
+
+    /**
+     * @return mixed
```
Comment (stefandoorn) [inline src/Entity/EstimatedArrivalTrait.php]: What about these type hints? Might be old, but might also be nice to adjust them to string where applicable.

**Comment by ryross** [comment]
Thanks for the push to clean up those type hints :)

Updated the `build` calls as well.

**Comment by ryross** [comment]
Wondering if opening this temporarily will show those updates?

**Comment by ryross** [comment]
Doesn't look like it.

**Comment by ryross** [comment]
Weird that StyleCI is failing. Closing this again, waiting for @mazzarito to accept the PR

**Pull request closed without merging**